python/semantic_kernel/data/text_search/utils.py

```diff
@@ -44,30 +48,24 @@ def create_options(
         SearchOptions: The options.
 
     """
+    new_options = options_class()
     if options:
         if not isinstance(options, options_class):
+            inputs = None
             try:
-                # Validate the options in one go
-                new_options = options_class.model_validate(
-                    options.model_dump(exclude_none=True, exclude_defaults=True, exclude_unset=True),
-                )
-            except ValidationError:
-                # if that fails, go one by one
-                new_options = options_class()
-                for key, value in options.model_dump(
-                    exclude_none=True, exclude_defaults=True, exclude_unset=True
-                ).items():
-                    setattr(new_options, key, value)
+                inputs = options.model_dump(exclude_none=True, exclude_defaults=True, exclude_unset=True)
+            except Exception:
+                logger.warning("Options are not valid. Creating new options.")
+            if inputs:
+                new_options = options_class.model_validate(inputs)
         else:
             new_options = options
         for key, value in kwargs.items():
             if key in new_options.model_fields:
                 setattr(new_options, key, value)
     else:
-        try:
+        with suppress(ValidationError):
```

Comment:
Question: With this suppressed, will `new_options` become `None` or something else when an exception happens?

Reply:
It is already defined earlier as a new object